tuna/celery_app/celery_workers.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python3
+###############################################################################
+#
+# MIT License
+#
+# Copyright (c) 2024 Advanced Micro Devices, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+###############################################################################
+"""Interface class to set up and launch tuning functionality"""
+import os
+import logging
+import subprocess
+
+from tuna.utils.logger import setup_logger
+
+LOGGER: logging.Logger = setup_logger('celery_workers')
+
+
+def launch_worker_per_node(machines, cmd, formatted=False):
+  """Launch celery worker for compile"""
+  final_cmd = cmd
+  subp_list = []
+  for machine in machines:
+    try:
+      if formatted:
```

Comment:
@cderb  can we remove the formatted bool? there is no else part, what is the intention for that?

Reply:
I didn't add this portion. replace should function correctly weather HOSTNAME is present or not.